runtime/src/accounts_db.rs

```diff
@@ -1303,6 +1332,7 @@ impl AccountsDB {
 
     pub fn include_owner_in_hash(slot: Slot) -> bool {
         // Account hashing updated to include owner activates at this slot on the testnet
+        // DANGER: Remove this guard ABSOLUTELY before the mainnet-beta transitions to v1.1.
```

Comment:
FYI: @mvines this is free-riding comment. Should separate from this pr..? If you prefer, I'll fine to do: b0478b4b8045eab0d9e7de33a39389a1159f9ea1

I've just noticed this pitfall whiling thinking about debut of eager rent collection.

Reply:
Thanks for pointing this out.   Landing it in this PR is fine.
I've created https://github.com/solana-labs/solana/projects/54 so we can start tracking any potential issues with upgrading mainnet-beta from 1.0 to 1.1, and added a note about this in that project.